cds-snc/notification-api
# feat: disable Pinpoint, use only SNS for SMS

This PR:
- removes disable Pinpoint as an SMS provider
- clean up code related to Pinpoint
- sends SMS in another region when an SMS is sent with a custom sender

Because we'll be running with only a single SMS provider for a few weeks/months, some features are not relevant/don't work anymore. This is the case for the automatic switch of SMS providers in case a provider is down/slow. I did not delete the logic/tests for now as we really want to avoid having a single SMS provider for too long so I skipped these tests.

If you want to test this feature locally, you'll need to:
- run this branch locally
- add an inbound number in platform admin (I can give you a long code number on Slack)
- attach this inbound number to a service you own
- run the Celery SMS worker `scripts/run_celery_sms.sh`
- send a text, see that it's sent in us-west-2 with a specific phone number

I'm happy to share my screen and to demo this.

Infrastructure work done beforehand:
https://github.com/cds-snc/notification-terraform/pull/126
https://github.com/cds-snc/notification-terraform/pull/128
https://github.com/cds-snc/notification-terraform/pull/131

Trello card https://trello.com/c/pTa9pwcR/215-clean-up-amazon-pinpoint
